### PR TITLE
Add curve preference option to configuration files

### PR DESCRIPTION
**Pull request by ColinSullivan1**
* Add a curve_preference parameter to specify/order curve preferences.  e.g. to serve many connections, one could specify "CurveP256" in order to reduce CPU usage.
* Add details to tls help
* Deprecated older 1.4 cipher suite code

### CODE REVIEW

File: server/ciphersuites_1.4.go [hunk 1]
```diff
@@ -1,33 +0,0 @@
-// Copyright 2015 Apcera Inc. All rights reserved.
```
Comment (derekcollison) [inline server/ciphersuites_1.4.go]: So does this change now require at least 1.5 to build? If so, totally ok, but should update main README.

Comment (derekcollison) [inline server/ciphersuites_1.4.go]: Also should we add in the new ones from 1.8?

Comment (ColinSullivan1) [inline server/ciphersuites_1.4.go]: I'll update the README.  I was debating on including 1.8 - it is beta today, but scheduled for release on 1/31/17 -  wdyt?

Comment (derekcollison) [inline server/ciphersuites_1.4.go]: Let's add in 1.8 support. Thanks.

Comment (ColinSullivan1) [inline server/ciphersuites_1.4.go]: Done...  I didn't add 1.8beta2 compilation for travis, but build and tests pass locally.  Shall I add that at this point, or do we want to wait for the 1.8 release?

**Comment by derekcollison** [comment]
Let's go ahead and add 1.8 for Travis if we have 1.8 specific code.

File: server/ciphersuites_1.4.go (hunk 1 above)
Comment (ColinSullivan1) [inline server/ciphersuites_1.4.go]: Derek, thanks for the guidance on 1.8 beta - we should be all set now.

**Pull request merged**